### ISSUE TITLE
Seeing some errors using this plugin

### ISSUE THREAD

**Issue by patflynn** [b3er/idea-archive-browser#1]
Super useful plugin! Seems to have an error though. See below:

update failed for AnAction with ID=MainToolBar: com.github.b3er.idea.plugins.arc.browser.ZipArchiveStructureProvider.getData(Ljava/util/Collection;Ljava/lang/String;)Ljava/lang/Object;

```
update failed for AnAction with ID=MainToolBar: com.github.b3er.idea.plugins.arc.browser.ZipArchiveStructureProvider.getData(Ljava/util/Collection;Ljava/lang/String;)Ljava/lang/Object;
java.lang.AbstractMethodError: com.github.b3er.idea.plugins.arc.browser.ZipArchiveStructureProvider.getData(Ljava/util/Collection;Ljava/lang/String;)Ljava/lang/Object;
	at com.intellij.ide.util.treeView.AbstractTreeStructureBase.getDataFromProviders(AbstractTreeStructureBase.java:102)
	at com.intellij.ide.projectView.impl.AbstractProjectViewPane.getData(AbstractProjectViewPane.java:291)
	at com.intellij.ide.projectView.impl.ProjectViewImpl$MyPanel.getData(ProjectViewImpl.java:1041)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:81)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:69)
	at com.intellij.ide.impl.DataManagerImpl.access$300(DataManagerImpl.java:55)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.a(DataManagerImpl.java:391)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.getData(DataManagerImpl.java:362)
	at com.intellij.openapi.actionSystem.DataContext.getData(DataContext.java:59)
	at com.intellij.openapi.actionSystem.AnActionEvent.getData(AnActionEvent.java:165)
	at com.intellij.openapi.actionSystem.AnActionEvent.getProject(AnActionEvent.java:120)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:104)
	at com.intellij.openapi.actionSystem.impl.Utils.a(Utils.java:188)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:120)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:96)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.a(ActionToolbarImpl.java:942)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.access$000(ActionToolbarImpl.java:65)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl$1.updateActionsImpl(ActionToolbarImpl.java:174)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater$MyUpdateRunnable.run(ToolbarUpdater.java:186)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.a(FocusManagerImpl.java:649)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.f(FocusManagerImpl.java:630)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.p(FocusManagerImpl.java:600)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.b(FocusManagerImpl.java:482)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:2493)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:462)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater.a(ToolbarUpdater.java:108)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater.access$400(ToolbarUpdater.java:40)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater$MyTimerListener.run(ToolbarUpdater.java:158)
	at com.intellij.openapi.actionSystem.impl.WeakTimerListener.run(WeakTimerListener.java:54)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.a(ActionManagerImpl.java:1414)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.a(ActionManagerImpl.java:1404)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.actionPerformed(ActionManagerImpl.java:1388)
	at javax.swing.Timer.fireActionPerformed(Timer.java:313)
	at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:679)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)


java.lang.AbstractMethodError: com.github.b3er.idea.plugins.arc.browser.ZipArchiveStructureProvider.getData(Ljava/util/Collection;Ljava/lang/String;)Ljava/lang/Object;
	at com.intellij.ide.util.treeView.AbstractTreeStructureBase.getDataFromProviders(AbstractTreeStructureBase.java:102)
	at com.intellij.ide.projectView.impl.AbstractProjectViewPane.getData(AbstractProjectViewPane.java:291)
	at com.intellij.ide.projectView.impl.ProjectViewImpl$MyPanel.getData(ProjectViewImpl.java:1041)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:81)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:69)
	at com.intellij.ide.impl.DataManagerImpl.access$300(DataManagerImpl.java:55)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.a(DataManagerImpl.java:391)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.getData(DataManagerImpl.java:362)
	at com.intellij.openapi.actionSystem.DataContext.getData(DataContext.java:59)
	at com.intellij.openapi.actionSystem.AnActionEvent.getData(AnActionEvent.java:165)
	at com.intellij.openapi.actionSystem.AnActionEvent.getProject(AnActionEvent.java:120)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performDumbAwareUpdate(ActionUtil.java:104)
	at com.intellij.openapi.actionSystem.impl.Utils.a(Utils.java:188)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:120)
	at com.intellij.openapi.actionSystem.impl.Utils.expandActionGroup(Utils.java:96)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.a(ActionToolbarImpl.java:942)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl.access$000(ActionToolbarImpl.java:65)
	at com.intellij.openapi.actionSystem.impl.ActionToolbarImpl$1.updateActionsImpl(ActionToolbarImpl.java:174)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater$MyUpdateRunnable.run(ToolbarUpdater.java:186)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.a(FocusManagerImpl.java:649)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.f(FocusManagerImpl.java:630)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.p(FocusManagerImpl.java:600)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.b(FocusManagerImpl.java:482)
	at com.intellij.util.ui.UIUtil.invokeLaterIfNeeded(UIUtil.java:2493)
	at com.intellij.openapi.wm.impl.FocusManagerImpl.doWhenFocusSettlesDown(FocusManagerImpl.java:462)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater.a(ToolbarUpdater.java:108)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater.access$400(ToolbarUpdater.java:40)
	at com.intellij.openapi.actionSystem.impl.ToolbarUpdater$MyTimerListener.run(ToolbarUpdater.java:158)
	at com.intellij.openapi.actionSystem.impl.WeakTimerListener.run(WeakTimerListener.java:54)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.a(ActionManagerImpl.java:1414)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.a(ActionManagerImpl.java:1404)
	at com.intellij.openapi.actionSystem.impl.ActionManagerImpl$MyTimer.actionPerformed(ActionManagerImpl.java:1388)
	at javax.swing.Timer.fireActionPerformed(Timer.java:313)
	at javax.swing.Timer$DoPostEvent.run(Timer.java:245)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:679)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)


java.lang.AbstractMethodError: com.github.b3er.idea.plugins.arc.browser.ZipArchiveStructureProvider.getData(Ljava/util/Collection;Ljava/lang/String;)Ljava/lang/Object;
	at com.intellij.ide.util.treeView.AbstractTreeStructureBase.getDataFromProviders(AbstractTreeStructureBase.java:102)
	at com.intellij.ide.projectView.impl.AbstractProjectViewPane.getData(AbstractProjectViewPane.java:291)
	at com.intellij.ide.projectView.impl.ProjectViewImpl$MyPanel.getData(ProjectViewImpl.java:1041)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:81)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:69)
	at com.intellij.ide.impl.DataManagerImpl.access$300(DataManagerImpl.java:55)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.a(DataManagerImpl.java:391)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.getData(DataManagerImpl.java:368)
	at com.intellij.openapi.actionSystem.DataKey.getData(DataKey.java:75)
	at com.intellij.openapi.command.impl.FocusBasedCurrentEditorProvider.getCurrentEditor(FocusBasedCurrentEditorProvider.java:32)
	at com.intellij.openapi.command.impl.UndoManagerImpl.d(UndoManagerImpl.java:305)
	at com.intellij.openapi.command.impl.UndoManagerImpl.a(UndoManagerImpl.java:274)
	at com.intellij.openapi.command.impl.UndoManagerImpl.a(UndoManagerImpl.java:231)
	at com.intellij.openapi.command.impl.UndoManagerImpl.access$000(UndoManagerImpl.java:60)
	at com.intellij.openapi.command.impl.UndoManagerImpl$1.commandStarted(UndoManagerImpl.java:177)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.a(CoreCommandProcessor.java:361)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.a(CoreCommandProcessor.java:148)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:109)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.c(EditorImpl.java:5858)
	at com.intellij.openapi.editor.impl.EditorImpl$MyMouseAdapter.mousePressed(EditorImpl.java:5788)
	at java.awt.Component.processMouseEvent(Component.java:6530)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4522)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:675)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)


java.lang.AbstractMethodError: com.github.b3er.idea.plugins.arc.browser.ZipArchiveStructureProvider.getData(Ljava/util/Collection;Ljava/lang/String;)Ljava/lang/Object;
	at com.intellij.ide.util.treeView.AbstractTreeStructureBase.getDataFromProviders(AbstractTreeStructureBase.java:102)
	at com.intellij.ide.projectView.impl.AbstractProjectViewPane.getData(AbstractProjectViewPane.java:291)
	at com.intellij.ide.projectView.impl.ProjectViewImpl$MyPanel.getData(ProjectViewImpl.java:1041)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:81)
	at com.intellij.ide.impl.DataManagerImpl.a(DataManagerImpl.java:69)
	at com.intellij.ide.impl.DataManagerImpl.access$300(DataManagerImpl.java:55)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.a(DataManagerImpl.java:391)
	at com.intellij.ide.impl.DataManagerImpl$MyDataContext.getData(DataManagerImpl.java:362)
	at com.intellij.openapi.actionSystem.DataKey.getData(DataKey.java:75)
	at com.intellij.ui.SpeedSearchBase.a(SpeedSearchBase.java:548)
	at com.intellij.ui.SpeedSearchBase.access$100(SpeedSearchBase.java:55)
	at com.intellij.ui.SpeedSearchBase$2.focusLost(SpeedSearchBase.java:91)
	at java.awt.AWTEventMulticaster.focusLost(AWTEventMulticaster.java:230)
	at java.awt.AWTEventMulticaster.focusLost(AWTEventMulticaster.java:229)
	at java.awt.AWTEventMulticaster.focusLost(AWTEventMulticaster.java:229)
	at java.awt.Component.processFocusEvent(Component.java:6425)
	at java.awt.Component.processEvent(Component.java:6289)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1954)
	at java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:995)
	at java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:661)
	at com.intellij.ide.IdeKeyboardFocusManager.dispatchEvent(IdeKeyboardFocusManager.java:27)
	at java.awt.Component.dispatchEventImpl(Component.java:4760)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at com.intellij.ide.IdeEventQueue.b(IdeEventQueue.java:843)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:679)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:391)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

**Comment by b3er** [comment]
Thanks, will be fixed in 0.6

**Comment by patflynn** [comment]
:+1:
